modules/markup/external/external.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package external
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup"
+	"code.gitea.io/gitea/modules/setting"
+
+	gouuid "github.com/satori/go.uuid"
+)
+
+// RegisterParsers registers all supported third part parsers according settings
+func RegisterParsers() {
+	for _, render := range setting.ExternalMarkupRenders {
+		if render.Enabled && render.Command != "" && len(render.FileExtensions) > 0 {
+			markup.RegisterParser(&Parser{render})
+		}
+	}
+}
+
+// Parser implements markup.Parser for orgmode
+type Parser struct {
+	setting.MarkupRender
+}
+
+// Name implements markup.Parser
+func (p *Parser) Name() string {
+	return p.MarkupName
+}
+
+// Extensions implements markup.Parser
```

Comment:
Rewrite comments please, it should tell what is the function doing.

Reply:
done.